sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -297,6 +298,22 @@ public void setDecimal(int ordinal, Decimal value, int precision) {
     }
   }
 
+  @Override
+  public void setInterval(int ordinal, CalendarInterval value) {
+    assertIndexIsValid(ordinal);
+    long cursor = getLong(ordinal) >>> 32;
+    assert cursor > 0 : "invalid cursor " + cursor;
+    if (value == null) {
```

Comment:
shall we also zero-out the bytes if it's null?

Reply:
Methinks only if we directly call `getInterval` api, otherwise this won't hurt. WDYT?

Reply:
Methinks only if we directly call `getInterval` API, otherwise, this won't hurt. WDYT?

Reply:
It's important to keep the binary format deterministic. If an interval field is null, it's better to guarantee that the value portion is all 0, rather than random values.

We can only rely on binary equality check if the binary format is deterministic.

Reply:
OK, let zero-out them to keep safe.